state-chain/runtime/src/chainflip.rs

```diff
@@ -819,11 +821,17 @@ impl OnBroadcastReady<Bitcoin> for BroadcastReadyProvider {
 				let output_len = outputs.len();
 				let vout = output_len - 1;
 				let change_output = outputs.get(vout).unwrap();
-				Environment::add_bitcoin_change_utxo(
-					change_output.amount,
-					UtxoId { tx_id, vout: vout as u32 },
-					batch_transfer.change_utxo_key,
-				);
+				if ScriptPubkey::Taproot(batch_transfer.change_utxo_key) ==
+					change_output.script_pubkey
+				{
+					Environment::add_bitcoin_change_utxo(
```

Comment:
We should also consider checking against the previous_pubkey: imagine we send a tx before but witness it after a rotation/key handover. 

Also: Maybe we should do this check for each of the outputs? 

For example imagine someone sends funds to the vault (I don't know why...)

ie we could do:

```rust
for output in outputs {
	if [vault.current_key, vault.previous_key].contains(output.address) {
		// add to utxos
	}
}
```


@martin-chainflip ?

Reply:
Alternatively: we encode the change amount directly in the BatchTransfer type, then we don't need to parse anything from the utxos. 

(ie. we change the type to:

```rust
pub struct BatchTransfer {
	pub bitcoin_transaction: BitcoinTransaction,
	// Use a struct:
	pub change_utxo_details: (
		[u8; 32], VOut, TxId, Amount
	),
}
```

Reply:
i think Dan's first suggestions is best.

Reply:
Anyone can open a channel sending funds to the vault. (could be by accident, could be on purpose: maybe we want to top up the vault?)

Reply:
But why would that change the order of the outputs of the tx? When we create a btc tx, we select some utxos from the vault, and create output utxos (transfers, and possibly change utxo appended at the end), and save it this output utxo list in the ```BitcoinTransaction``` type. So when the tx is successfully signed, we then take the change utxo and add it to the available utxo list which is what we are doing here. I dont see how someone sending funds to the vault is relevant here.

Reply:
Imagine I request a swap with the BTC vault as the destination address. Then we construct a transaction with two outputs, both of which will go into the vault. Pretty theoretical example, but this code clearly does the right thing, whereas just checking the last output adds a lot of unnecessary assumptions that might break in the future.

Reply:
In that theoretical example, i would say that the right thing to do is not egress at all because you are basically taking a utxo from the vault, splitting it in two and sending it back to the vault (which is pointless if not actually harmful because we are splitting utxos) and so instead of allowing the egress to happen and deal with it here in an indirect way, I would just not egress.

The code here is specifically to add the change utxo to the available utxo list and is tightly coupled with the code of creating btc tx. I dont think we are making assumptions here because we construct the btc tx a certain way (putting the change utxo at the end) and until we change how we construct btc tx, this should always be the case.